InAppUtils/InAppUtils.m

```diff
@@ -117,42 +161,35 @@ - (void) doPurchaseProduct:(NSString *)productIdentifier
                 payment.applicationUsername = username;
             }
             [[SKPaymentQueue defaultQueue] addPayment:payment];
-            _callbacks[RCTKeyForInstance(payment.productIdentifier)] = callback;
         } else {
-            callback(@[@"invalid_product"]);
+            NSLog(@"invalid_product"); // TODO we lose feedback here, how do we solve this? Send error event?
         }
     } @catch (NSException *exception) {
-        callback(@[@"purchase_product_exception"]);
+        NSLog(@"purchase_product_exception"); // TODO we lose feedback here, how do we solve this? Send error event?
     }
 }
 
 - (void)paymentQueue:(SKPaymentQueue *)queue
 restoreCompletedTransactionsFailedWithError:(NSError *)error
 {
-    NSString *key = RCTKeyForInstance(@"restoreRequest");
-    RCTResponseSenderBlock callback = _callbacks[key];
-    if (callback) {
+    if (hasListeners) { // Only send events if anyone is listening
```

Comment:
restore functies etc. hebben we niet nodig right? dus dat is in principe gewoon om module compleet te maken?

Reply:
Heb so far alleen deze module omgeschreven naar listeners